docs/synapse/adminguides/usersroles.rstorm

```diff
@@ -0,0 +1,268 @@
+
+**Example:**
+
+Add the new role "cattribution analyst":
+
+.. storm-cli:: auth.role.add "cattribution analyst"
+
+.. TIP::
+  
+  Roles are represented by a unique 128-bit identifier (iden). You can later change information about the role
+  (such as the role name) without affecting the underlying role or any associated permissions or users.
+
+
+.. _admin_role_show:
+
+Display a Role
+--------------
+
+The :ref:`storm-auth-role-show` command displays information about a role, including any assigned rules
+(permissions) and their associated objects.
+
+**Example:**
+
+Display information for the "all" role:
+
+.. storm-cli:: auth.role.show all
+
+
+.. _admin_role_mod:
+
+Modify a Role
+-------------
+
+The :ref:`storm-auth-role-mod` command modifies a role. The command can be used to change the name of the
+role.
+
+**Example:**
+
+Change the name of the role "cattribution analyst" to "meow-ware analyst":
+
+.. storm-cli:: auth.role.mod "cattribution analyst" --name "meow-ware analyst"
+
+
+.. _admin_role_list:
+
+List all Roles
+--------------
+
+The :ref:`storm-auth-role-list` command lists all roles in the Cortex.
+
+**Example:**
+
+List all roles:
+
+.. storm-pre:: auth.role.add "a-cat-emic researcher"
+.. storm-pre:: auth.role.add "cattribution analyst"
+.. storm-cli:: auth.role.list
+
+
+.. _admin_role_del:
+
+Delete a Role
+-------------
+
+The :ref:`storm-auth-role-del` command deletes a role.
+
+**Example:**
+
+Delete the role "meow-ware analyst":
+
+.. storm-cli:: auth.role.del "meow-ware analyst"
+
+.. NOTE::
+  
+  Deleting a role has no impact on any users who have been granted the role (other than losing any permissions
+  provided by that role). The user accounts remain intact and the role is simply removed from each user's list
+  of roles.
+
+
+.. _admin_grant_roles:
+
+Grant or Revoke Roles
+=====================
+
+**Granting** a role to a user allows the user to inherit the role's permissions. **Revoking** a role removes the
+associated permissions from the user. It is not possible to grant a role to another role (i.e., roles cannot be
+nested).
+
+Roles can be granted or revoked using the :ref:`storm-auth-user-grant` and :ref:`storm-auth-user-revoke` commands.
+
+**Examples:**
+
+Grant the role "cattribution analyst" to the user "ron":
+
+.. storm-cli:: auth.user.grant ron "cattribution analyst"
+
+Revoke the role "a-cat-emic researcher" from user "ron":
+
+.. storm-pre:: auth.user.grant ron "a-cat-emic researcher"
+
+.. storm-cli:: auth.user.revoke ron "a-cat-emic researcher"
+
+
+.. NOTE::
+  
+  The order in which roles are granted to a user matters; when determining whether a user has permission to
+  perform an action, the permissions for each of the user's roles are checked in sequence.
+  
+  Each role granted to a user is added to the **end** of the set of roles. To "reorder" roles, you must revoke
+  them and grant them in the desired order (unless using the commercial Synapse UI, which allows drag-and-drop
```

Comment:
https://synapse.docs.vertex.link/en/latest/synapse/autodocs/stormtypes_prims.html#setroles-idens is used to set an arbitrary list of roles, which reordering does

Reply:
Noted. Made similar update wrt rules (for users and roles).